src/main/java/com/frameworkium/core/ui/tests/BaseTest.java

```diff
@@ -49,6 +49,40 @@
     /** Logger for subclasses (logs with correct class i.e. not BaseTest). */
     protected final Logger logger = LogManager.getLogger(this);
 
+    /** Pool of drivers to be reused. **/
+    private static BlockingQueue<Driver> driverPool;
+
+    /**
+     * Method which runs before the test suite to initialise a pool of thread drivers,
+     * if the REUSE_BROWSER property is defined. If it is not, the default behaviour of a driver
+     * per run will be followed.
+     */
+    @BeforeSuite
+    public static void initialiseDriverPool() {
+        if (Property.REUSE_BROWSER.isSpecified()) {
+            int threads = (Property.THREAD.isSpecified()) ? Integer.parseInt(Property.THREAD.getValue()) : 1;
+            driverPool = new ArrayBlockingQueue(threads);
+            for (int i = 0; i < threads; i++) {
+                driverPool.add(new DriverSetup().instantiateDriver());
+            }
+        }
+    }
+
+    /**
+     * Method which returns the next available driver from the pool.
+     * The pool should always have at least one, as it has been initialised with one driver
+     * per thread and each thread pushed its driver back to the pool, when it completes a test.
+     * <p>In the unlikely scenario that the previous execution didn't add the driver back to the pool
+     * for whatever reason, then create a new driver to return.</p>
+     * @return Driver the next available driver from the pool of drivers
+     */
+    public static Driver getNextAvailableDriverFromPool() {
+        if (driverPool.isEmpty()) {
+            driverPool.add(new DriverSetup().instantiateDriver());
```

Comment:
Would the following also work?

```java
if (driverPool.isEmpty()) {
    return new DriverSetup().instantiateDriver();
}
return driverPool.remove();
```

It feels safer to limit the number of interactions with `driverPool`.

Reply:
Done.